pkg/asset/installconfig/gcp/dns.go

```diff
@@ -0,0 +1,28 @@
+package gcp
+
+import (
+	"golang.org/x/net/context"
+	"google.golang.org/api/dns/v1"
+)
+
+// GetPublicZone returns a DNS managed zone from the provided project which matches the baseDomain
+func GetPublicZone(baseDomain, project string) (managedZone *dns.ManagedZone, err error) {
+	ctx := context.Background()
+	//TODO: Determine if this will be proper/acceptable way to invoke services with credentials
```

Comment:
https://github.com/openshift/installer/blob/45f81e1d950cbc80e46e0b230e11b0032de8c868/pkg/asset/installconfig/aws/basedomain.go#L90
I think it's fine.